src/parse.ts

```diff
@@ -22,8 +22,8 @@ function getEntityBaseTypes(entityType, entityTypes) {
   const baseTypes = [];
 
   while (entityType) {
-    const baseTypeName = typeNameFromType(entityType['$']['BaseType'])
-    entityType = entityTypes.find(entity => entity['$']['Name'] == baseTypeName);
+    const baseTypeName = typeNameFromType(entityType._attributes.BaseType);
```

Comment:
A lot of the `'$'` to `_attributes` can be avoided by tweaking the `attributesKey` value on the conversion library.  Why didn't you do that?

Reply:
Well, i dont really understand how to do this. Could you please explain? Also will this tweak add any unnecessary overhead?

Reply:
I'm saying if you change 
[`const metadata = xml2js.xml2js(xml, { compact: true, trim: true, alwaysArray: true });`](https://github.com/elasticio/odata2openapi/blob/xml-memory-fix/src/parse.ts#L316) 
to 
`const metadata = xml2js.xml2js(xml, { compact: true, trim: true, alwaysArray: true, attributesKey: '$'  });` 
most of the code changes in this file become unnecessary.

Reply:
Fixed